pilot/pkg/config/kube/ingress/conversion_test.go

```diff
@@ -21,10 +21,123 @@ import (
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 
+	networking "istio.io/api/networking/v1alpha3"
+
 	meshconfig "istio.io/api/mesh/v1alpha1"
 	"istio.io/istio/pilot/pkg/model"
 )
 
+func TestConversion(t *testing.T) {
+	ingress := v1beta1.Ingress{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Namespace: "mock", // goes into backend full name
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "my.host.com",
```

Comment:
define this as const and use here and below.

Reply:
I think it also makes things harder to read. A string is immutable too.

Reply:
maybe, but it's prone to typos. compiler cannot check that you are using the right string. 
(actually myHostName doesn't seem less readable to me than "my.host.com"). 
anyway, it's not going to lead to a tragic error in this file so not a biggie, up to you.

Reply:
I'm not writing code for the compiler :-) The proper way to do this test is to move it to a file and 
read the file - in which case there will be no constant. I wrote it this way only because I am not sure what future this code has and it may not be worth adding too many more tests for ingress if we'll get rid of it soon... 

The test was mostly to help debugging and fixing the code - actual coverage will be in the test env (separate PR updating the stability cluster - and someone will need to bring back ingress tests in simple e2e).